src/main/java/com/ericsson/ei/waitlist/WaitListWorker.java

```diff
@@ -65,20 +66,24 @@ public void run() {
         if(shutdownInProgress) {
             return;
         }
-        List<String> documents = waitListStorageHandler.getWaitList();
-        for (String document : documents) {
-            try {
-                ObjectMapper objectMapper = new ObjectMapper();
-                JsonNode eventJson = objectMapper.readTree(document);
-                String id = eventJson.get("_id").asText();
-                if (eventToObjectMapHandler.isEventInEventObjectMap(id)) {
-                    waitListStorageHandler.dropDocumentFromWaitList(document);
-                } else {
-                    checkTargetAggregationsExistAndRepublishEvent(eventJson);
+        try {
+            List<String> documents = waitListStorageHandler.getWaitList();
```

Comment:
Please extract all logic in the try catch blocks to it´s own methods.
This regards not only the new try/catch you created but please refactor the already existing code to it´s own methods.

Basicly a try/catch should only contain a methods call and possible a return statement. And a try/catch should be the only logic in a method. This renders several methods, but makes it easier to read.

If you wish to read more about good Java code a tip is to study the book Clean Code by Robert C. Martin

Most parts of Eiffel Intelligence was written before we studied this book and learnt a better way to write code, thus there is a lot of refactoring to be done.